### PR TITLE
feat: add api rename code

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

add rename code api

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

add api rename code

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

**Comment by Peefy** [comment]
@CodiumAI-Agent /review

**Comment by Peefy** [comment]
I think the `path` and `source_codes` in all functions are confusing and need to be readjusted or merged into one parameter.

File: kclvm/tools/src/LSP/src/rename.rs [hunk 1]
```diff
@@ -137,17 +241,18 @@ fn apply_text_edit(edit: &TextEdit, line: &str) -> String {
 pub fn rename_symbol(
     pkg_root: &str,
     file_paths: &[String],
+    source_codes: HashMap<String, String>,
     symbol_path: &str,
     new_name: String,
 ) -> Result<HashMap<Url, Vec<TextEdit>>> {
     // 1. from symbol path to the symbol
-    let symbol_spec = parse_symbol_selector_spec(pkg_root, symbol_path)?;
+    let symbol_spec = parse_symbol_selector_spec(pkg_root, symbol_path, source_codes.clone())?;
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/rename.rs:249]: Prevent the deep clone function calling.

**Comment by Peefy** [comment]
@CodiumAI-Agent /review

File: kclvm/tools/src/LSP/src/rename.rs [hunk 2]
```diff
@@ -1,50 +1,200 @@
...
+use parking_lot::RwLock;
+use ra_ap_vfs::{Vfs, VfsPath};
 use std::fs;
 use std::path::PathBuf;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
 
-/// Service for renaming all the occurrences of the target symbol in the files. This API will rewrite files if they contain symbols to be renamed.
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/rename.rs:15]: Why remove comments here?

File: kclvm/tools/src/LSP/src/rename.rs [hunk 3]
```diff
@@ -1,50 +1,200 @@
 use crate::{
     from_lsp::kcl_pos,
     goto_def::find_def_with_gs,
-    util::{build_word_index_for_file_paths, parse_param_and_compile, read_file, Param},
+    util::{build_word_index_for_source_codes, VirtualLocation},
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/rename.rs:4]: Windows CI failed. :(

File: kclvm/tools/src/LSP/src/rename.rs [hunk 4]
```diff
@@ -1,50 +1,200 @@
...
-fn apply_rename_changes(changes: &HashMap<Url, Vec<TextEdit>>) -> Result<HashMap<String, String>> {
-    let mut result = HashMap::new();
-    for (url, edits) in changes {
-        let file_content = read_file(
-            &url.to_file_path()
-                .map_err(|_| anyhow!("Failed to convert URL to file path"))?
-                .display()
-                .to_string(),
-        )?;
+pub fn rename_symbol_on_code(
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/rename.rs:53]: Would U pls add more documents?

File: kclvm/parser/src/lib.rs [hunk 5]
```diff
@@ -344,7 +346,8 @@ impl Loader {
         let mut pkgs = HashMap::new();
         let workdir = compile_entries.get_root_path().to_string();
 
-        debug_assert_eq!(compile_entries.len(), self.paths.len());
+        // debug_assert_eq!(compile_entries.len(), self.paths.len());
+        // todo: need to be removed after support vfs
```
Comment (Peefy) [inline kclvm/parser/src/lib.rs:350]: Remove un-used code.

Comment (zong-zhe) [reply kclvm/parser/src/lib.rs]: fixed

File: kclvm/tools/src/LSP/src/rename.rs (hunk 4 above)
Comment (zong-zhe) [reply kclvm/tools/src/LSP/src/rename.rs:57]: fixed.

File: kclvm/tools/src/LSP/src/rename.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/tools/src/LSP/src/rename.rs:15]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**